roles/rsyslog/tasks/main_core.yml

```
@@ -37,42 +62,23 @@
       vars:
         __rsyslog_files_to_remove: "{{ __rsyslog_confs.stdout_lines }}"
 
-# if inputs are given and rsyslog_enabled is true, rsyslog.conf will be
-# overwritten, so no need to reinstall package
-- name: Reinstall package to restore rsyslog config if purging
-  when:
-    - logging_purge_confs | bool | d(false)
-    - not (rsyslog_inputs and __rsyslog_enabled | bool)
-    - not __logging_is_ostree | d(false)
-  block:
-    # it is assumed that the only packages providing config files that might
-    # be modified are the rsyslog related base packages (but not others like iproute);
-    # if this is not so, then additional packages will need to be added to this check
-    - name: Get status of rsyslog packages
-      command: rpm -V {{ item }}  # noqa command-instead-of-module
-      loop: "{{ __rsyslog_base_packages | select('search', 'rsyslog') | list }}"
-      register: __rsyslog_package_status
-      failed_when: false
-      changed_when: false
-
-    # this cannot be done with package: and not even with dnf:
-    - name: Reset original confs - reinstall logging package
+    # this cannot be done with package: and not even with dnf
+    - name: Reinstall required packages
       changed_when: true
+      #  FIXME
+      #  name: "{{ __rsyslog_base_packages + (__rsyslog_tls_packages
+      #    if (logging_pki_files | d([]) | length > 0) else []) +
+      #    (rsyslog_extra_packages | flatten) }}"
       shell: |
         {{ ansible_pkg_mgr }} reinstall -y "{{ __rsyslog_base_packages | select('search', 'rsyslog') | join(' ') }}"
```

```suggestion
        {{ ansible_pkg_mgr }} reinstall -y {{ __rsyslog_only_packages | join(' ') }}
```
I don't think you need the double quotes around the list of packages

---

can't use __rsyslog_only_packages (unless you want to define it elsewhere)

but the issue with double quotes applies - that is, I don't think `dnf reinstall -y "rsyslog rsyslog-omfile ... rsyslog-elasticsearch"` will work?  Does `dnf` parse the argument into separate packages?